internal/config/cnimgr/cnimgr.go

```diff
@@ -90,3 +99,31 @@ func (c *CNIManager) Shutdown() {
 		watcher <- false
 	}
 }
+
+// GC calls the plugin's GC to clean up any resources concerned with stale pods
+// (pod other than the ones provided by validPodList). The call to the plugin
+// will be deferred until it is ready logging any errors then and returning nil
+// error here.
+func (c *CNIManager) GC(ctx context.Context, validPodList PodNetworkLister) error {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+	if c.lastError != nil {
+		c.validPodList = validPodList
+		return nil
+	}
+	return c.doGC(ctx, validPodList)
```

Comment:
Yeah, we can. I was just being overly jealous I imagine not setting it if it was not going to be used (because the plugin was actually ready). Will change it around.